src/test/java/org/junitpioneer/jupiter/ExpectedToFailExtensionTests.java

```diff
@@ -101,6 +101,16 @@ void failsOnWorkingTest() {
 				.hasMessage("Test marked as 'expected to fail' succeeded; remove @ExpectedToFail from it");
 	}
 
+	@Test
+	void failsOnTestThrowingUnexpectedException() {
+		ExecutionResults results = PioneerTestKit.executeTestMethod(ExpectedToFailTestCases.class, "unexpectedException");
+		assertThat(results)
+				.hasSingleStartedTest()
+				.whichFailed()
+				.withExceptionInstanceOf(AssertionError.class)
+				.hasMessage("Test marked as 'expected to fail' failed with an unexpected class java.lang.IllegalArgumentException exception");
```

Comment:
I would not bake in the classname to the test, but that's personal preference.

The exception is actually an [AssertionFailedError](https://ota4j-team.github.io/opentest4j/docs/1.0.0/api/org/opentest4j/AssertionFailedError.html?is-external=true) an extension of but not an `AssertionError` (exactly).